dspace-api/src/main/java/org/dspace/app/launcher/Readme.md

```diff
@@ -0,0 +1,57 @@
+# Adding new scripts to the dspace CLI 
+
+Scripts can be added to the script launcher using 2 mechanisms
+
+1. Defining a script in [dspace]/config/launcher.xml
+2. Creating a spring bean of type _org.dspace.app.launcher.CommandType_
+
+The advantage of the second mechanism is that a dependency can provide both the script and add it to the launcher without needing to modify the configuration of the receiving party.
+
```

Comment:
with regards to the configuration file. In essence you just need to provide a spring bean of type CommandType. The system doesn't really care how you do it, it leverages DSpace's existing spring infrastructure. To recap that one: create a file named _spring-dspace-*-services.xml_ in the /spring folder of any jar on the classpath and put your beans in => they will be picked up. You could also put a file in _config/spring/api_ of your receiving DSpace but that would defeat this contribution's purpose.